curl_cffi/requests/session.py

```diff
@@ -335,97 +208,7 @@ def __init__(
-        curl.setopt(CurlOpt.SSLVERSION, curl_tls_version | CurlSslVersion.MAX_DEFAULT)
-        assert curl_tls_version == CurlSslVersion.TLSv1_2, "Only TLS v1.2 works for now."
-
-        cipher_names = []
-        for cipher in ciphers.split("-"):
-            cipher_id = int(cipher)
-            cipher_name = TLS_CIPHER_NAME_MAP[cipher_id]
-            cipher_names.append(cipher_name)
-
-        curl.setopt(CurlOpt.SSL_CIPHER_LIST, ":".join(cipher_names))
-
-        if extensions.endswith("-21"):
-            extensions = extensions[:-3]
-            warnings.warn(
-                "Padding(21) extension found in ja3 string, whether to add it should "
-                "be managed by the SSL engine. The TLS client hello packet may contain "
-                "or not contain this extension, any of which should be correct.",
-                stacklevel=1,
-            )
-        extension_ids = set(int(e) for e in extensions.split("-"))
-        self._toggle_extensions_by_ids(curl, extension_ids)
-
-        if not permute:
-            curl.setopt(CurlOpt.TLS_EXTENSION_ORDER, extensions)
-
-        curve_names = []
-        for curve in curves.split("-"):
-            curve_id = int(curve)
-            curve_name = TLS_EC_CURVES_MAP[curve_id]
-            curve_names.append(curve_name)
-
-        curl.setopt(CurlOpt.SSL_EC_CURVES, ":".join(curve_names))
-
-        assert int(curve_formats) == 0, "Only curve_formats == 0 is supported."
-
-    def _set_akamai_options(self, curl, akamai: str):
-        """
-        Detailed explanation: https://www.blackhat.com/docs/eu-17/materials/eu-17-Shuster-Passive-Fingerprinting-Of-HTTP2-Clients-wp.pdf
-        """
-        settings, window_update, streams, header_order = akamai.split("|")
-
-        # For compatiblity with tls.peet.ws
-        settings = settings.replace(",", ";")
-
-        curl.setopt(CurlOpt.HTTP_VERSION, CurlHttpVersion.V2_0)
-
-        curl.setopt(CurlOpt.HTTP2_SETTINGS, settings)
-        curl.setopt(CurlOpt.HTTP2_WINDOW_UPDATE, int(window_update))
-
-        if streams != "0":
-            curl.setopt(CurlOpt.HTTP2_STREAMS, streams)
-
-        # m,a,s,p -> masp
-        # curl-impersonate only accepts masp format, without commas.
-        curl.setopt(CurlOpt.HTTP2_PSEUDO_HEADERS_ORDER, header_order.replace(",", ""))
-
-    def _set_extra_fp(self, curl, fp: ExtraFingerprints):
-        if fp.tls_signature_algorithms:
-            curl.setopt(CurlOpt.SSL_SIG_HASH_ALGS, ",".join(fp.tls_signature_algorithms))
-
-        curl.setopt(CurlOpt.SSLVERSION, fp.tls_min_version | CurlSslVersion.MAX_DEFAULT)
-        curl.setopt(CurlOpt.TLS_GREASE, int(fp.tls_grease))
-        curl.setopt(CurlOpt.SSL_PERMUTE_EXTENSIONS, int(fp.tls_permute_extensions))
-        curl.setopt(CurlOpt.SSL_CERT_COMPRESSION, fp.tls_cert_compression)
-        curl.setopt(CurlOpt.STREAM_WEIGHT, fp.http2_stream_weight)
-        curl.setopt(CurlOpt.STREAM_EXCLUSIVE, fp.http2_stream_exclusive)
-
-    def _set_curl_options(
```

Comment:
The entire function should be moved to `options.py`, I guess.

Reply:
Yeah it was probably for the better too since this file was getting quite large